Sources/Sentry/SentryOptions.m

```diff
@@ -59,16 +59,19 @@ @implementation SentryOptions {
     // SentryCrashIntegration needs to be initialized before SentryAutoSessionTrackingIntegration.
     // And SentrySessionReplayIntegration before SentryCrashIntegration.
     NSMutableArray<Class> *defaultIntegrations = [NSMutableArray<Class> arrayWithObjects:
-#if SENTRY_HAS_UIKIT && !TARGET_OS_VISION
+#if SENTRY_TARGET_REPLAY_SUPPORTED
             [SentrySessionReplayIntegration class],
-#endif
+#endif // SENTRY_TARGET_REPLAY_SUPPORTED
         [SentryCrashIntegration class],
 #if SENTRY_HAS_UIKIT
         [SentryAppStartTrackingIntegration class], [SentryFramesTrackingIntegration class],
-        [SentryPerformanceTrackingIntegration class], [SentryScreenshotIntegration class],
-        [SentryUIEventTrackingIntegration class], [SentryViewHierarchyIntegration class],
+        [SentryPerformanceTrackingIntegration class], [SentryUIEventTrackingIntegration class],
+        [SentryViewHierarchyIntegration class],
         [SentryWatchdogTerminationTrackingIntegration class],
 #endif // SENTRY_HAS_UIKIT
+#if SENTRY_TARGET_REPLAY_SUPPORTED
+        [SentryScreenshotIntegration class],
+#endif // SENTRY_TARGET_REPLAY_SUPPORTED
```

Comment:
`SENTRY_TARGET_REPLAY_SUPPORTED` means `SENTRY_HAS_UIKIT && !TARGET_OS_VISION`... Seems like we should pick one of these:

1. rename `SENTRY_TARGET_REPLAY_SUPPORTED`
2. create a new macro that (for now) is defined the same way
3. use `SENTRY_HAS_UIKIT && !TARGET_OS_VISION` at these code sites instead of overloading the meaning of that macro

Reply:
I'd be curious to know why we can't do this on vision os?

Reply:
> I'd be curious to know why we can't do this on vision os? Is there some logic in eg UIRedactBuilder or SentryViewPhotographer that could be more granularly conditionally compiled between visionOS or iOS/tvOS?

We cant use `#if os(visionOS)` in swift yet because we support Xcode 14

Reply:
> 1. rename SENTRY_TARGET_REPLAY_SUPPORTED

This

Reply:
> We cant use #if os(visionOS) in swift yet because we support Xcode 14

I was asking what API we are using that aren't available on visionOS.

Reply:
There is no api restriction, it is only the compiler directive that does not work and breaks compilation.